docs/src/components/productX/XTreeViewDemo.tsx

```diff
@@ -1,261 +1,276 @@
+    getGroupTransitionProps,
+    status,
+    publicAPI,
+  } = useTreeItem2({ id, itemId, children, label, disabled, rootRef: ref });
+
+  const item = publicAPI.getItem(itemId);
+  const expandable = isExpandable(children);
+  let icon;
+  if (expandable) {
+    if (status.expanded) {
+      icon = FolderOpenRounded;
+    } else {
+      icon = FolderRounded;
+    }
+  } else if (item.fileType) {
+    icon = getIconFromFileType(item.fileType);
+  }
+
+  return (
+    <TreeItem2Provider itemId={itemId}>
+      <StyledTreeItemRoot {...getRootProps(other)}>
+        <CustomTreeItemContent
+          {...getContentProps({
+            className: clsx('content', {
+              'Mui-expanded': status.expanded,
+              'Mui-selected': status.selected,
+              'Mui-focused': status.focused,
+              'Mui-disabled': status.disabled,
+            }),
+          })}
+        >
+          {expandable && (
+            <TreeItem2IconContainer {...getIconContainerProps()}>
+              <TreeItem2Icon status={status} />
+            </TreeItem2IconContainer>
+          )}
+
+          <CustomLabel {...getLabelProps({ icon, expandable })} />
+        </CustomTreeItemContent>
+        {children && <TransitionComponent {...getGroupTransitionProps()} />}
+      </StyledTreeItemRoot>
+    </TreeItem2Provider>
+  );
 });
 
 const code = `
-<TreeView
-  aria-label="file system navigator"
-  defaultExpanded={['1', '1.1', '1.2', '2', '2.3']}
-  sx={{ height: { xs: 260, sm: 460 }, overflowY: 'auto', p: 1 }}
->
-  <TreeItem nodeId="1" label="Drive">
-    <TreeItem nodeId="1.1" label="Backup">
-      <TreeItem
-        nodeId="1.1.1"
-        label="Jan 2021.pdf"
-        ContentProps={{ lastNestedChild: true }}
-      />
-      <TreeItem
-        nodeId="1.1.2"
-        label="Feb 2021.pdf"
-        ContentProps={{ lastNestedChild: true }}
-      />
-      <TreeItem
-        nodeId="1.1.3"
-        label="Mar 2021.pdf"
-        ContentProps={{ lastNestedChild: true }}
-      />
-    </TreeItem>
-    <TreeItem nodeId="1.2" label="Photos">
-      <TreeItem
-        nodeId="1.2.1"
-        label="family.jpeg"
-        ContentProps={{ lastNestedChild: true }}
-      />
-      <TreeItem
-        nodeId="1.2.2"
-        label="my_dogpng"
-        ContentProps={{ lastNestedChild: true }}
-      />
-    </TreeItem>
-  </TreeItem>
-  <TreeItem nodeId="2" label="Favorite">
-    <TreeItem
-      nodeId="2.1"
-      label="MUI_retreat_photo.jpg"
-      ContentProps={{ lastNestedChild: true }}
-    />
-    <TreeItem
-      nodeId="2.2"
-      label="v6_secrets.mkv"
-      ContentProps={{ lastNestedChild: true }}
-    />
-    <TreeItem nodeId="2.3" label="Other pictures">
-      <TreeItem
-        nodeId="2.3.1"
-        label="my_avatar.jpg"
-        ContentProps={{ lastNestedChild: true }}
-      />
-    </TreeItem>
-  </TreeItem>
-</TreeView>`;
+<RichTreeView
+  items={ITEMS}
+  aria-label="file explorer"
+  defaultExpandedItems={['1', '1.1']}
```

Comment:
```suggestion
  aria-label="File explorer"
  defaultExpandedItems={['1', '1.1', '1.2', '2']}
```

I'd keep a few more items opened for optical alignment with the column on the left + capitalize the aria-label